components/LanguageSwitcher/src/index.tsx

```diff
@@ -0,0 +1,65 @@
+import React from 'react';
+import LanguageSwitcher from './LanguageSwitcher';
+import './index.scss';
+import LanguageSwitcherTitle from './LanguageSwitcherTitle';
+import LanguageSwitcherList from './LanguageSwitcherList';
+import LanguageSwitcherListItem from './LanguageSwitcherListItem';
+import { Link } from '@gemeente-denhaag/link';
+import { ArrowRightIcon, CheckedIcon } from '@gemeente-denhaag/icons';
+import clsx from 'clsx';
+
+interface Language {
+  title: string;
+  url?: string;
+  lang?: string;
+  hrefLang?: string;
+  active?: boolean;
+}
+
+export interface LanguageSwitcherLogicProps {
+  title: string;
+  languages: Array<Language>;
+  mobileMenuScrolled?: boolean;
+}
+
+export const LanguageSwitcherLogic = ({
+  title = '',
+  languages = [],
+  mobileMenuScrolled,
+}: LanguageSwitcherLogicProps) => {
+  const languageListContent = languages.map((language, key) => {
+    const classNames = clsx(
+      'denhaag-language-switcher__list-item-link',
+      language.active ? 'denhaag-language-switcher__list-item-link--active' : '',
+    );
+
+    return (
+      <LanguageSwitcherListItem key={key}>
+        <Link
+          tabIndex={mobileMenuScrolled ? -1 : undefined}
+          {...(!language.active && { icon: <ArrowRightIcon />, iconAlign: 'start' })}
+          {...(language.active && { icon: <CheckedIcon />, iconAlign: 'end' })}
+          href={language.url}
+          hrefLang={language.hrefLang}
+          className={classNames}
+        >
+          {language.title}
+        </Link>
+      </LanguageSwitcherListItem>
+    );
+  });
+
+  return (
+    <LanguageSwitcher>
+      <LanguageSwitcherTitle className="utrecht-heading-4">{title}</LanguageSwitcherTitle>
```

Comment:
I'm still not sure if adding a classname of a specific heading is a great idea or will cause headaches in the future as one would not expect this component to change necessarily when changing the properties of heading-4...
Okay for now, but something we might want to change later